# Introduction

In this report, we will investigate the vegeatables that provide the most nutrition.

In [55]:
import pandas as pd
import numpy as np
import matplotlib 
import seaborn
import matplotlib as plt
import math
import plotly.express as px

In [4]:
fruit_and_veg = pd.read_csv("fruit-and-vegetable-consumption-in-california-residents-20122013.csv")

In [5]:
fruit_and_veg.head()

Year           Age Group             Category    Type  \
0  2012  Adolescent (12-17)                 All    Total   
1  2012  Adolescent (12-17)               Gender    Male   
2  2012  Adolescent (12-17)               Gender  Female   
3  2012  Adolescent (12-17)  Gender by Age- Male   12-13   
4  2012  Adolescent (12-17)  Gender by Age- Male   14-15   

   Five or more servings of fruits and vegetables (%)  
0                                               40.2   
1                                               37.3   
2                                               43.3   
3                                               35.2   
4                                               42.1

In [6]:
fruit_and_veg.rename(columns = {'Five or more servings of fruits and vegetables (%)':'Servings of F/V => 5'}, inplace = True)

In [7]:
fruit_and_veg

Year           Age Group             Category                   Type  \
0   2012  Adolescent (12-17)                 All                   Total   
1   2012  Adolescent (12-17)               Gender                   Male   
2   2012  Adolescent (12-17)               Gender                 Female   
3   2012  Adolescent (12-17)  Gender by Age- Male                  12-13   
4   2012  Adolescent (12-17)  Gender by Age- Male                  14-15   
..   ...                 ...                  ...                    ...   
65  2013        Child (6-11)     Parent Education  Some College/Graduate   
66  2013        Child (6-11)    Physical Activity           > 60 minutes   
67  2013        Child (6-11)    Physical Activity           < 60 minutes   
68  2013        Child (6-11)    Overweight Status         Not Overweight   
69  2013        Child (6-11)    Overweight Status       Overweight/Obese   

    Servings of F/V => 5  
0                   40.2  
1                   37.3  
2                   43.3  
3                   35.2  
4                   42.1  
..                   ...  
65                  24.4  
66                  29.9  
67                  23.6  
68                  29.8  
69                  26.9  

[70 rows x 5 columns]

In [8]:
average_servings_over5 = fruit_and_veg["Servings of F/V => 5"].mean()
average_servings_over5


36.75000000000001

I found the average here and now know that across all groups, the average number of people who eat over 5 servings of fruit or vegtable a day is about 37%

In [9]:
select_AgeGroup_Adolescent = fruit_and_veg.loc[fruit_and_veg['Age Group'] == "Adolescent (12-17)"]
select_AgeGroup_Adolescent

Year           Age Group                Category                     Type  \
0   2012  Adolescent (12-17)                    All                     Total   
1   2012  Adolescent (12-17)                  Gender                     Male   
2   2012  Adolescent (12-17)                  Gender                   Female   
3   2012  Adolescent (12-17)     Gender by Age- Male                    12-13   
4   2012  Adolescent (12-17)     Gender by Age- Male                    14-15   
5   2012  Adolescent (12-17)     Gender by Age- Male                    16-17   
6   2012  Adolescent (12-17)   Gender by Age- Female                    12-13   
7   2012  Adolescent (12-17)   Gender by Age- Female                    14-15   
8   2012  Adolescent (12-17)   Gender by Age- Female                    16-17   
9   2012  Adolescent (12-17)          Race/Ethnicity                    White   
10  2012  Adolescent (12-17)          Race/Ethnicity         African American   
11  2012  Adolescent (12-17)          Race/Ethnicity                   Latino   
12  2012  Adolescent (12-17)          Race/Ethnicity              Asian/Other   
13  2012  Adolescent (12-17)                  Income        Less than $15,000   
14  2012  Adolescent (12-17)                  Income         $15,000 - 24,999   
15  2012  Adolescent (12-17)                  Income         $25,000 - 34,999   
16  2012  Adolescent (12-17)                  Income         $35,000 - 49,999   
17  2012  Adolescent (12-17)                  Income         $50,000 - 74,999   
18  2012  Adolescent (12-17)                  Income       $75,000 or greater   
19  2012  Adolescent (12-17)       Physical Activity                  Regular   
20  2012  Adolescent (12-17)       Physical Activity                Irregular   
21  2012  Adolescent (12-17)       Overweight Status           Not Overweight   
22  2012  Adolescent (12-17)       Overweight Status         Overweight/Obese   
23  2012  Adolescent (12-17)  CalFresh Status, % FPL              Participant   
24  2012  Adolescent (12-17)  CalFresh Status, % FPL  Likely Eligible, ≤ 130%   
25  2012  Adolescent (12-17)  CalFresh Status, % FPL    Not Eligible, > 185%    

    Servings of F/V => 5  
0                   40.2  
1                   37.3  
2                   43.3  
3                   35.2  
4                   42.1  
5                   34.6  
6                   50.1  
7                   39.8  
8                   40.5  
9                   40.3  
10                  35.2  
11                  44.2  
12                  29.6  
13                  43.2  
14                  38.2  
15                  54.1  
16                  41.0  
17                  29.7  
18                  42.3  
19                  41.9  
20                  34.4  
21                  41.2  
22                  39.1  
23                  47.5  
24                  41.9  
25                  36.2

In [10]:
select_AgeGroup_child = fruit_and_veg.loc[fruit_and_veg['Age Group'] == "Child (6-11)"]
select_AgeGroup_child

Year     Age Group           Category                   Type  \
54  2013  Child (6-11)               All                   Total   
55  2013  Child (6-11)             Gender                   Male   
56  2013  Child (6-11)             Gender                 Female   
57  2013  Child (6-11)                Age                    6-8   
58  2013  Child (6-11)                Age                   9-11   
59  2013  Child (6-11)     Race/Ethnicity                  White   
60  2013  Child (6-11)     Race/Ethnicity       African American   
61  2013  Child (6-11)     Race/Ethnicity                 Latino   
62  2013  Child (6-11)     Race/Ethnicity            Asian/Other   
63  2013  Child (6-11)   Parent Education  Less than High School   
64  2013  Child (6-11)   Parent Education   High School Graduate   
65  2013  Child (6-11)   Parent Education  Some College/Graduate   
66  2013  Child (6-11)  Physical Activity           > 60 minutes   
67  2013  Child (6-11)  Physical Activity           < 60 minutes   
68  2013  Child (6-11)  Overweight Status         Not Overweight   
69  2013  Child (6-11)  Overweight Status       Overweight/Obese   

    Servings of F/V => 5  
54                  27.1  
55                  24.0  
56                  30.3  
57                  28.3  
58                  25.6  
59                  27.3  
60                  24.5  
61                  27.5  
62                  25.9  
63                  27.9  
64                  30.0  
65                  24.4  
66                  29.9  
67                  23.6  
68                  29.8  
69                  26.9

In [11]:
fruit_and_veg.iloc[26:53]

Year    Age Group                Category  \
26  2013  Adult (18+)                    All    
27  2013  Adult (18+)                  Gender   
28  2013  Adult (18+)                  Gender   
29  2013  Adult (18+)                     Age   
30  2013  Adult (18+)                     Age   
31  2013  Adult (18+)                     Age   
32  2013  Adult (18+)                     Age   
33  2013  Adult (18+)                     Age   
34  2013  Adult (18+)          Race/Ethnicity   
35  2013  Adult (18+)          Race/Ethnicity   
36  2013  Adult (18+)          Race/Ethnicity   
37  2013  Adult (18+)          Race/Ethnicity   
38  2013  Adult (18+)               Education   
39  2013  Adult (18+)               Education   
40  2013  Adult (18+)               Education   
41  2013  Adult (18+)               Education   
42  2013  Adult (18+)                  Income   
43  2013  Adult (18+)                  Income   
44  2013  Adult (18+)                  Income   
45  2013  Adult (18+)                  Income   
46  2013  Adult (18+)                  Income   
47  2013  Adult (18+)       Physical Activity   
48  2013  Adult (18+)       Physical Activity   
49  2013  Adult (18+)       Overweight Status   
50  2013  Adult (18+)       Overweight Status   
51  2013  Adult (18+)  CalFresh Status, % FPL   
52  2013  Adult (18+)  CalFresh Status, % FPL   

                                   Type  Servings of F/V => 5  
26                                Total                  39.8  
27                                 Male                  34.3  
28                               Female                  45.1  
29                              18 - 24                  36.2  
30                              25 - 34                  43.3  
31                              35 - 50                  38.7  
32                              51 - 64                  40.6  
33                                  65+                  39.5  
34                                White                  40.1  
35                     African American                  38.5  
36                               Latino                  40.1  
37                          Asian/Other                  38.8  
38                Less than High School                  34.9  
39                 High School Graduate                  36.9  
40                         Some College                  37.1  
41                     College Graduate                  48.5  
42                    Less than $15,000                  40.1  
43                     $15,000 - 24,999                  40.7  
44                     $25,000 - 34,999                  37.0  
45                     $35,000 - 49,999                  33.0  
46                             $50,000+                  41.2  
47           Met Aerobic Recommendation                  45.1  
48  Did Not Meet Aerobic Recommendation                  30.8  
49                       Not Overweight                  42.6  
50                     Overweight/Obese                  38.5  
51                          Participant                  40.1  
52              Likely Eligible, ≤ 130%                  35.0

In [12]:
select_AgeGroup_adult = fruit_and_veg.loc[fruit_and_veg['Age Group'] == "Adult (18+)"]
select_AgeGroup_adult

Year    Age Group                Category  \
26  2013  Adult (18+)                    All    
27  2013  Adult (18+)                  Gender   
28  2013  Adult (18+)                  Gender   
29  2013  Adult (18+)                     Age   
30  2013  Adult (18+)                     Age   
31  2013  Adult (18+)                     Age   
32  2013  Adult (18+)                     Age   
33  2013  Adult (18+)                     Age   
34  2013  Adult (18+)          Race/Ethnicity   
35  2013  Adult (18+)          Race/Ethnicity   
36  2013  Adult (18+)          Race/Ethnicity   
37  2013  Adult (18+)          Race/Ethnicity   
38  2013  Adult (18+)               Education   
39  2013  Adult (18+)               Education   
40  2013  Adult (18+)               Education   
41  2013  Adult (18+)               Education   
42  2013  Adult (18+)                  Income   
43  2013  Adult (18+)                  Income   
44  2013  Adult (18+)                  Income   
45  2013  Adult (18+)                  Income   
46  2013  Adult (18+)                  Income   
47  2013  Adult (18+)       Physical Activity   
48  2013  Adult (18+)       Physical Activity   
49  2013  Adult (18+)       Overweight Status   
50  2013  Adult (18+)       Overweight Status   
51  2013  Adult (18+)  CalFresh Status, % FPL   
52  2013  Adult (18+)  CalFresh Status, % FPL   
53  2013  Adult (18+)  CalFresh Status, % FPL   

                                   Type  Servings of F/V => 5  
26                                Total                  39.8  
27                                 Male                  34.3  
28                               Female                  45.1  
29                              18 - 24                  36.2  
30                              25 - 34                  43.3  
31                              35 - 50                  38.7  
32                              51 - 64                  40.6  
33                                  65+                  39.5  
34                                White                  40.1  
35                     African American                  38.5  
36                               Latino                  40.1  
37                          Asian/Other                  38.8  
38                Less than High School                  34.9  
39                 High School Graduate                  36.9  
40                         Some College                  37.1  
41                     College Graduate                  48.5  
42                    Less than $15,000                  40.1  
43                     $15,000 - 24,999                  40.7  
44                     $25,000 - 34,999                  37.0  
45                     $35,000 - 49,999                  33.0  
46                             $50,000+                  41.2  
47           Met Aerobic Recommendation                  45.1  
48  Did Not Meet Aerobic Recommendation                  30.8  
49                       Not Overweight                  42.6  
50                     Overweight/Obese                  38.5  
51                          Participant                  40.1  
52              Likely Eligible, ≤ 130%                  35.0  
53                Not Eligible, > 185%                   39.9

I've seperated this dataframe so now I have three different variables that contain information on a single age group. Within that age group, I have different categories and can more easily move through this information

What I would like to do with this data. 
1. compare income and veg within age groups and then graph it
2. compare education with veg 
3. compare overweight status 
4. compare ages within these groups to veg 
5. compare gender of all groups and veg consumption 

In [13]:
!pip install scipy
!pip install statsmodels

In [14]:
import scipy 
import statsmodels

In [15]:
from scipy import stats
from statsmodels.stats.weightstats import DescrStatsW

In [16]:
fruit_and_veg.sample(10)

Year           Age Group                Category                   Type  \
3   2012  Adolescent (12-17)     Gender by Age- Male                  12-13   
66  2013        Child (6-11)       Physical Activity           > 60 minutes   
41  2013         Adult (18+)               Education       College Graduate   
16  2012  Adolescent (12-17)                  Income       $35,000 - 49,999   
4   2012  Adolescent (12-17)     Gender by Age- Male                  14-15   
68  2013        Child (6-11)       Overweight Status         Not Overweight   
25  2012  Adolescent (12-17)  CalFresh Status, % FPL  Not Eligible, > 185%    
44  2013         Adult (18+)                  Income       $25,000 - 34,999   
21  2012  Adolescent (12-17)       Overweight Status         Not Overweight   
46  2013         Adult (18+)                  Income               $50,000+   

    Servings of F/V => 5  
3                   35.2  
66                  29.9  
41                  48.5  
16                  41.0  
4                   42.1  
68                  29.8  
25                  36.2  
44                  37.0  
21                  41.2  
46                  41.2

In [18]:
adult_max_serv = np.max(select_AgeGroup_adult["Servings of F/V => 5"])
adult_max_serv

48.5

In [19]:
adult_min_serv = np.min(select_AgeGroup_adult["Servings of F/V => 5"])
adult_min_serv

30.8

In [20]:
range_adult_serv = np.ptp(select_AgeGroup_adult["Servings of F/V => 5"])
range_adult_serv

17.7

In [37]:
adult_serv = select_AgeGroup_adult["Servings of F/V => 5"]
adult_serv.head(10)

26    39.8
27    34.3
28    45.1
29    36.2
30    43.3
31    38.7
32    40.6
33    39.5
34    40.1
35    38.5
Name: Servings of F/V => 5, dtype: float64

In [23]:
adult_sorted_serv = select_AgeGroup_adult.sort_values(by=["Servings of F/V => 5"], ascending=True).reset_index(drop=True)
adult_sorted_serv.head(10)

Year    Age Group                Category  \
0  2013  Adult (18+)       Physical Activity   
1  2013  Adult (18+)                  Income   
2  2013  Adult (18+)                  Gender   
3  2013  Adult (18+)               Education   
4  2013  Adult (18+)  CalFresh Status, % FPL   
5  2013  Adult (18+)                     Age   
6  2013  Adult (18+)               Education   
7  2013  Adult (18+)                  Income   
8  2013  Adult (18+)               Education   
9  2013  Adult (18+)       Overweight Status   

                                  Type  Servings of F/V => 5  
0  Did Not Meet Aerobic Recommendation                  30.8  
1                     $35,000 - 49,999                  33.0  
2                                 Male                  34.3  
3                Less than High School                  34.9  
4              Likely Eligible, ≤ 130%                  35.0  
5                              18 - 24                  36.2  
6                 High School Graduate                  36.9  
7                     $25,000 - 34,999                  37.0  
8                         Some College                  37.1  
9                     Overweight/Obese                  38.5

In [25]:
adult_serv_mean = np.mean(select_AgeGroup_adult["Servings of F/V => 5"])
adult_serv_mean

39.157142857142865

In [26]:
adult_serv_median = np.median(select_AgeGroup_adult["Servings of F/V => 5"])
adult_serv_median

39.65

# Children (6-11)
1. compare parent education with veg
2. compare overweight status
3. compare gender of all groups and veg consumption

In [48]:
select_AgeGroup_child["Category"].unique()

array(['All ', 'Gender', 'Age', 'Race/Ethnicity', 'Parent Education',
       'Physical Activity', 'Overweight Status'], dtype=object)

In [50]:
child_serv_min = np.min(select_AgeGroup_child["Servings of F/V => 5"])
child_serv_min

23.6

In [51]:
child_serv_max = np.max(select_AgeGroup_child["Servings of F/V => 5"])
child_serv_max

30.3

In [52]:
select_AgeGroup_child["Servings of F/V => 5"].unique()

array([27.1, 24. , 30.3, 28.3, 25.6, 27.3, 24.5, 27.5, 25.9, 27.9, 30. ,
       24.4, 29.9, 23.6, 29.8, 26.9])

In [53]:
select_AgeGroup_child["Servings of F/V => 5"].nunique()

16

In [58]:
child_parent_ed = select_AgeGroup_child.loc[(select_AgeGroup_child["Category"] == "Parent Education")]
child_parent_ed

Year     Age Group          Category                   Type  \
63  2013  Child (6-11)  Parent Education  Less than High School   
64  2013  Child (6-11)  Parent Education   High School Graduate   
65  2013  Child (6-11)  Parent Education  Some College/Graduate   

    Servings of F/V => 5  
63                  27.9  
64                  30.0  
65                  24.4

In [65]:
fig1 = px.bar(child_parent_ed, x="Type", color="Servings of F/V => 5")
fig1.show()

Interestingly, in this study we find that children who have parents with a High School diploma eat more veggies than children whose parents have less education AND more education. 

In [67]:
child_weight = select_AgeGroup_child.loc[(select_AgeGroup_child["Category"] == "Overweight Status")]
child_weight

Year     Age Group           Category              Type  \
68  2013  Child (6-11)  Overweight Status    Not Overweight   
69  2013  Child (6-11)  Overweight Status  Overweight/Obese   

    Servings of F/V => 5  
68                  29.8  
69                  26.9

Children in this study who were overweight/obese typically eat 2.9% less fruits or veggies a day than the children who are considered 'not overweight'. 

In [70]:
child_gender = select_AgeGroup_child.loc[(select_AgeGroup_child["Category"] == "Gender")]
child_gender

Year     Age Group Category    Type  Servings of F/V => 5
55  2013  Child (6-11)   Gender    Male                  24.0
56  2013  Child (6-11)   Gender  Female                  30.3

Here we also see that females of this age group eat more than 6% more fruits/veggies a day than boys of this age group. 

# Adolescent (12-17)
1. compare parent education with veg
2. compare overweight status
3. compare gender of all groups and veg consumption

In [71]:
select_AgeGroup_Adolescent

Year           Age Group                Category                     Type  \
0   2012  Adolescent (12-17)                    All                     Total   
1   2012  Adolescent (12-17)                  Gender                     Male   
2   2012  Adolescent (12-17)                  Gender                   Female   
3   2012  Adolescent (12-17)     Gender by Age- Male                    12-13   
4   2012  Adolescent (12-17)     Gender by Age- Male                    14-15   
5   2012  Adolescent (12-17)     Gender by Age- Male                    16-17   
6   2012  Adolescent (12-17)   Gender by Age- Female                    12-13   
7   2012  Adolescent (12-17)   Gender by Age- Female                    14-15   
8   2012  Adolescent (12-17)   Gender by Age- Female                    16-17   
9   2012  Adolescent (12-17)          Race/Ethnicity                    White   
10  2012  Adolescent (12-17)          Race/Ethnicity         African American   
11  2012  Adolescent (12-17)          Race/Ethnicity                   Latino   
12  2012  Adolescent (12-17)          Race/Ethnicity              Asian/Other   
13  2012  Adolescent (12-17)                  Income        Less than $15,000   
14  2012  Adolescent (12-17)                  Income         $15,000 - 24,999   
15  2012  Adolescent (12-17)                  Income         $25,000 - 34,999   
16  2012  Adolescent (12-17)                  Income         $35,000 - 49,999   
17  2012  Adolescent (12-17)                  Income         $50,000 - 74,999   
18  2012  Adolescent (12-17)                  Income       $75,000 or greater   
19  2012  Adolescent (12-17)       Physical Activity                  Regular   
20  2012  Adolescent (12-17)       Physical Activity                Irregular   
21  2012  Adolescent (12-17)       Overweight Status           Not Overweight   
22  2012  Adolescent (12-17)       Overweight Status         Overweight/Obese   
23  2012  Adolescent (12-17)  CalFresh Status, % FPL              Participant   
24  2012  Adolescent (12-17)  CalFresh Status, % FPL  Likely Eligible, ≤ 130%   
25  2012  Adolescent (12-17)  CalFresh Status, % FPL    Not Eligible, > 185%    

    Servings of F/V => 5  
0                   40.2  
1                   37.3  
2                   43.3  
3                   35.2  
4                   42.1  
5                   34.6  
6                   50.1  
7                   39.8  
8                   40.5  
9                   40.3  
10                  35.2  
11                  44.2  
12                  29.6  
13                  43.2  
14                  38.2  
15                  54.1  
16                  41.0  
17                  29.7  
18                  42.3  
19                  41.9  
20                  34.4  
21                  41.2  
22                  39.1  
23                  47.5  
24                  41.9  
25                  36.2

Just from looking at my data, I can see that there is a 2.1% difference in fruit/veggie consumption between not overweight teens and overweight teens. That doesn't seem highly significicant. However, there is a 7.5% difference in consumption between teens who don't regularly work out vs teens who do. 

In [72]:
teen_income = select_AgeGroup_Adolescent.loc[(select_AgeGroup_Adolescent["Category"] == "Income")]
teen_income

Year           Age Group Category                Type  \
13  2012  Adolescent (12-17)   Income   Less than $15,000   
14  2012  Adolescent (12-17)   Income    $15,000 - 24,999   
15  2012  Adolescent (12-17)   Income    $25,000 - 34,999   
16  2012  Adolescent (12-17)   Income    $35,000 - 49,999   
17  2012  Adolescent (12-17)   Income    $50,000 - 74,999   
18  2012  Adolescent (12-17)   Income  $75,000 or greater   

    Servings of F/V => 5  
13                  43.2  
14                  38.2  
15                  54.1  
16                  41.0  
17                  29.7  
18                  42.3

In [73]:
fig2 = px.bar(teen_income, x="Type", color="Servings of F/V => 5")
fig2.show()

The teens who eat the highest percentage of fruits/veggies have a family income of $25,000 - $34,999. 
The teens who eat the smallest percentage of fruits/veggies have a family income of $50,000 - $74,000. 

In [77]:
teen_gen_by_age = select_AgeGroup_Adolescent.iloc[3:9]
teen_gen_by_age

Year           Age Group               Category   Type  \
3  2012  Adolescent (12-17)    Gender by Age- Male  12-13   
4  2012  Adolescent (12-17)    Gender by Age- Male  14-15   
5  2012  Adolescent (12-17)    Gender by Age- Male  16-17   
6  2012  Adolescent (12-17)  Gender by Age- Female  12-13   
7  2012  Adolescent (12-17)  Gender by Age- Female  14-15   
8  2012  Adolescent (12-17)  Gender by Age- Female  16-17   

   Servings of F/V => 5  
3                  35.2  
4                  42.1  
5                  34.6  
6                  50.1  
7                  39.8  
8                  40.5

In [81]:
fig3 = px.scatter(teen_gen_by_age, x="Type", y="Servings of F/V => 5", color="Category")
fig3.show()

The teens in this study who eat the most veggies are females between the age of 12-13. The ones who eat the least are boys between the ages of 16-17. 

# Adult (18+)
1. compare education with veg
2. compare overweight status
3. compare gender of all groups and veg consumption

In [82]:
select_AgeGroup_adult

Year    Age Group                Category  \
26  2013  Adult (18+)                    All    
27  2013  Adult (18+)                  Gender   
28  2013  Adult (18+)                  Gender   
29  2013  Adult (18+)                     Age   
30  2013  Adult (18+)                     Age   
31  2013  Adult (18+)                     Age   
32  2013  Adult (18+)                     Age   
33  2013  Adult (18+)                     Age   
34  2013  Adult (18+)          Race/Ethnicity   
35  2013  Adult (18+)          Race/Ethnicity   
36  2013  Adult (18+)          Race/Ethnicity   
37  2013  Adult (18+)          Race/Ethnicity   
38  2013  Adult (18+)               Education   
39  2013  Adult (18+)               Education   
40  2013  Adult (18+)               Education   
41  2013  Adult (18+)               Education   
42  2013  Adult (18+)                  Income   
43  2013  Adult (18+)                  Income   
44  2013  Adult (18+)                  Income   
45  2013  Adult (18+)                  Income   
46  2013  Adult (18+)                  Income   
47  2013  Adult (18+)       Physical Activity   
48  2013  Adult (18+)       Physical Activity   
49  2013  Adult (18+)       Overweight Status   
50  2013  Adult (18+)       Overweight Status   
51  2013  Adult (18+)  CalFresh Status, % FPL   
52  2013  Adult (18+)  CalFresh Status, % FPL   
53  2013  Adult (18+)  CalFresh Status, % FPL   

                                   Type  Servings of F/V => 5  
26                                Total                  39.8  
27                                 Male                  34.3  
28                               Female                  45.1  
29                              18 - 24                  36.2  
30                              25 - 34                  43.3  
31                              35 - 50                  38.7  
32                              51 - 64                  40.6  
33                                  65+                  39.5  
34                                White                  40.1  
35                     African American                  38.5  
36                               Latino                  40.1  
37                          Asian/Other                  38.8  
38                Less than High School                  34.9  
39                 High School Graduate                  36.9  
40                         Some College                  37.1  
41                     College Graduate                  48.5  
42                    Less than $15,000                  40.1  
43                     $15,000 - 24,999                  40.7  
44                     $25,000 - 34,999                  37.0  
45                     $35,000 - 49,999                  33.0  
46                             $50,000+                  41.2  
47           Met Aerobic Recommendation                  45.1  
48  Did Not Meet Aerobic Recommendation                  30.8  
49                       Not Overweight                  42.6  
50                     Overweight/Obese                  38.5  
51                          Participant                  40.1  
52              Likely Eligible, ≤ 130%                  35.0  
53                Not Eligible, > 185%                   39.9

In [84]:
adult_ed = select_AgeGroup_adult.loc[(select_AgeGroup_adult["Category"] == "Education")]
adult_ed

Year    Age Group   Category                   Type  Servings of F/V => 5
38  2013  Adult (18+)  Education  Less than High School                  34.9
39  2013  Adult (18+)  Education   High School Graduate                  36.9
40  2013  Adult (18+)  Education           Some College                  37.1
41  2013  Adult (18+)  Education       College Graduate                  48.5

In [85]:
fig4 = px.bar(adult_ed, x="Type", color="Servings of F/V => 5")
fig4.show()

In [87]:
adult_age = select_AgeGroup_adult.loc[(select_AgeGroup_adult["Category"] == "Age")]
adult_age

Year    Age Group Category     Type  Servings of F/V => 5
29  2013  Adult (18+)      Age  18 - 24                  36.2
30  2013  Adult (18+)      Age  25 - 34                  43.3
31  2013  Adult (18+)      Age  35 - 50                  38.7
32  2013  Adult (18+)      Age  51 - 64                  40.6
33  2013  Adult (18+)      Age      65+                  39.5

In [89]:
fig5 = px.bar(adult_age, x="Type", color="Servings of F/V => 5")
fig5.show()